src/publish/mod.rs

```diff
@@ -0,0 +1,194 @@
+//! Types, traits, and functions necessary to publish messages using hedwig
+
+use crate::{Error, Topic, ValidatedMessage};
+use futures_util::{
+    ready,
+    stream::{self, Stream},
+};
+use pin_project::pin_project;
+use std::{
+    collections::BTreeMap,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+pub mod publishers;
+
+#[cfg(feature = "sink")]
+#[cfg_attr(docsrs, doc(cfg(feature = "sink")))]
+pub mod sink;
```

Comment:
Same here.

Reply:
the sinks feel separate enough to me that the module still makes sense. plus this helps to side-step the doc cfg problem I mentioned above

Reply:
To me `use hedwig::sink` would still uniquely indicate I'm working with pushing data into a library and therefore into services behind. But I think this is fine too.